### PR TITLE
HTBHF-2480 Made claimStatus private to claim, added updateClaimStatus method

### PR DESCRIPTION
**Pull request by YetAnotherMatt**

### CODE REVIEW

File: api/build.gradle [hunk 1]
```diff
@@ -71,7 +71,7 @@ dependencies {
     implementation "net.javacrumbs.shedlock:shedlock-spring:2.4.0"
     implementation "net.javacrumbs.shedlock:shedlock-provider-jdbc-template:2.4.0"
     implementation "com.vladmihalcea:hibernate-types-52:2.4.3"
-    implementation "org.javers:javers-spring-boot-starter-sql:5.5.0:"
+    implementation "org.javers:javers-spring-boot-starter-sql:5.7.7:"
```
Comment (YetAnotherMatt) [inline api/build.gradle]: An unrelated change. I got confused by the order of changes in Javers so updated it to while I was figuring it out. I decided it did no harm to be using the latest version.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/entity/Claim.java [hunk 2]
```diff
@@ -66,4 +70,18 @@
...
     private Claimant claimant;
+
+    public void updateClaimStatus(ClaimStatus newStatus) {
+        if (claimStatus != newStatus) {
+            claimStatusTimestamp = LocalDateTime.now();
+        }
+        claimStatus = newStatus;
+    }
+
+    public void updateEligibilityStatus(EligibilityStatus newStatus) {
```
Comment (YetAnotherMatt) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/entity/Claim.java]: We're not actually using this method yet (i.e. we're never updating eligibilityStatus).

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/MessageProcessorScheduler.java [hunk 3]
```diff
@@ -88,7 +88,7 @@ public void processPaymentMessages() {
         messageProcessor.processMessagesOfType(MessageType.MAKE_PAYMENT);
     }
 
-    @Scheduled(cron = EVERY_30_SECONDS)
+    @Scheduled(cron = "05/35 * * * * *") // every 30 seconds, offset from other schedules by 5 seconds to prevent clashes with CREATE_NEW_CARD for new claims
```
Comment (YetAnotherMatt) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/MessageProcessorScheduler.java]: Again, unrelated, but I saw that for new claims the CREATE_NEW_CARD and REPORT_CLAIM messages were always clashing and causing one of them to fail with a hibernate concurrent update exception. This offsets reporting new claims by 5 seconds to prevent that.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/entity/ClaimTest.java [hunk 4]
```diff
@@ -86,4 +92,58 @@ void shouldReturnTheSameIdIfOneIsSet() {
...
+        LocalDateTime originalTimestamp = LocalDateTime.now().minusHours(1);
+        Claim claim = Claim.builder()
+                .claimStatus(ClaimStatus.NEW)
+                .claimStatusTimestamp(originalTimestamp)
+                .build();
+        LocalDateTime now = LocalDateTime.now();
+
+        claim.updateClaimStatus(ClaimStatus.ACTIVE);
+
+        Assertions.assertThat(claim.getClaimStatusTimestamp()).isAfterOrEqualTo(now);
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/entity/ClaimTest.java]: This should just be `assertThat`, don't need/want the `Assertions` part in front of it.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/entity/ClaimTest.java [hunk 5]
```diff
@@ -86,4 +92,58 @@ void shouldReturnTheSameIdIfOneIsSet() {
         //Then
         Assertions.assertThat(id).isEqualTo(claim.getId());
     }
+
+    @Test
+    void shouldUpdateClaimStatusAndTimestamp() {
+        LocalDateTime originalTimestamp = LocalDateTime.now().minusHours(1);
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/entity/ClaimTest.java]: I think after the usage of constants for times in other tests, it would be good to create a constant in this test class called `ONE_HOUR_AGO` to use in all these tests, it'll be really clear then.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/entity/ClaimTest.java (hunk 4 above)
Comment (YetAnotherMatt) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/entity/ClaimTest.java]: Unfortunately, there are other assertions in this class that make `Assertions.` necessary.

Comment (YetAnotherMatt) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/entity/ClaimTest.java]: e.g. `assertThat(violations).hasSingleConstraintViolation`
